### PR TITLE
system: soc: Get console uart number from dts

### PR DESCRIPTION
**Pull request by marekmatej**
This will provide console number entirely from the DTS

Signed-off-by: Marek Matej <marek.matej@espressif.com>

### CODE REVIEW

**Comment by marekmatej** [comment]
@sylvioalves, @LucasTambor PTAL

**Comment by LucasTambor** [comment]
Neat solution for the UART_NUM problem! But there is one thing that makes me uncomfortable with this solution that comes back to Glauber's PR and I would like to bring back to discussion to hear from you. I don't think we should use the DT macros in hal_espressif. Maybe put them in soc.h or soc.c (we can use the CONFIG_ASSERT macro) under zephyr's repository maybe better suited for this kind of thing.

**Comment by marekmatej** [comment]
> Neat solution for the UART_NUM problem! But there is one thing that makes me uncomfortable with this solution that comes back to Glauber's PR and I would like to bring back to discussion to hear from you. I don't think we should use the DT macros in hal_espressif. Maybe put them in soc.h or soc.c (we can use the CONFIG_ASSERT macro) under zephyr's repository maybe better suited for this kind of thing.

Attempt to implement the IDF configuration layer in the Zephyr `soc` is here:
https://github.com/zephyrproject-rtos/hal_espressif/pull/153
https://github.com/zephyrproject-rtos/zephyr/pull/52137

**Comment by marekmatej** [comment]
PTAL folks

File: components/esp_system/port/soc/esp32/clk.c [hunk 1]
```diff
@@ -28,7 +28,8 @@
 #else
 #include "stubs.h"
 #define CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ   ESP_SOC_DEFAULT_CPU_FREQ_MHZ
-#endif
+#define CONFIG_ESP32_RTC_CLK_CAL_CYCLES ESP_SOC_DEFAULT_RTC_CLK_CAL_CYCLES
```
Comment (sylvioalves) [inline components/esp_system/port/soc/esp32/clk.c:31]: `#define CONFIG_ESP32_RTC_CLK_CAL_CYCLES ESP_SOC_DEFAULT_RTC_CLK_CAL_CYCLES` is already being defined in PM PR in Zephyr's upstream. Can you remove this entry from all `clk.c` files?

File: zephyr/esp_shared/include/stubs.h [hunk 2]
```diff
@@ -1,26 +1,48 @@
 /*
- * Copyright (c) 2021 Espressif Systems (Shanghai) Co., Ltd.
+ * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
```
Comment (sylvioalves) [inline zephyr/esp_shared/include/stubs.h:2]: You can keep `2021` in here as is.

**Review by sylvioalves — APPROVED** [review]

**Review by LucasTambor — APPROVED** [review]

**Pull request merged**